Github Notebook Ecological Genomics

## Author: Lily Shapiro 
### Affiliation: UVM 

#### E-mail contact: lily.shapiro@uvm.edu


##### Start Date: 2020-01-22
##### End Date: 2020-05-08
##### Project Descriptions:   





# Table of Contents:  
  * [Entry 1: 2020-01-13, Monday](#id-section1)
      * [Entry 2: 2020-01-14, Tuesday](#id-section2)
    * [Entry 3: 2020-01-15, Wednesday](#id-section3)
     * [Entry 4: 2020-01-16, Thursday](#id-section4)
       * [Entry 5: 2020-01-17, Friday](#id-section5)
       * [Entry 6: 2020-01-20, Monday](#id-section6)
      * [Entry 7: 2020-01-21, Tuesday](#id-section7)
    * [Entry 8: 2020-01-22, Wednesday](#id-section8)
     * [Entry 9: 2020-01-23, Thursday](#id-section9)
    * [Entry 10: 2020-01-24, Friday](#id-section10)
    * [Entry 11: 2020-01-27, Monday](#id-section11)
     * [Entry 12: 2020-01-28, Tuesday](#id-section12)
     * [Entry 13: 2020-01-29, Wednesday](#id-section13)
     * [Entry 14: 2020-01-30, Thursday](#id-section14)
    * [Entry 15: 2020-01-31, Friday](#id-section15)
    * [Entry 16: 2020-02-03, Monday](#id-section16)
     * [Entry 17: 2020-02-04, Tuesday](#id-section17)
    * [Entry 18: 2020-02-05, Wednesday](#id-section18)
    * [Entry 19: 2020-02-06, Thursday](#id-section19)
    * [Entry 20: 2020-02-07, Friday](#id-section20)
    * [Entry 21: 2020-02-10, Monday](#id-section21)
    * [Entry 22: 2020-02-11, Tuesday](#id-section22)
     * [Entry 23: 2020-02-12, Wednesday](#id-section23)
    * [Entry 24: 2020-02-13, Thursday](#id-section24)
    * [Entry 25: 2020-02-14, Friday](#id-section25)
    * [Entry 26: 2020-02-17, Monday](#id-section26)
     * [Entry 27: 2020-02-18, Tuesday](#id-section27)
     * [Entry 28: 2020-02-19, Wednesday](#id-section28)
     * [Entry 29: 2020-02-20, Thursday](#id-section29)
    * [Entry 30: 2020-02-21, Friday](#id-section30)
    * [Entry 31: 2020-02-24, Monday](#id-section31)
    * [Entry 32: 2020-02-25, Tuesday](#id-section32)
     * [Entry 33: 2020-02-26, Wednesday](#id-section33)
    * [Entry 34: 2020-02-27, Thursday](#id-section34)
    * [Entry 35: 2020-02-28, Friday](#id-section35)
    * [Entry 36: 2020-03-02, Monday](#id-section36)
     * [Entry 37: 2020-03-03, Tuesday](#id-section37)
     * [Entry 38: 2020-03-04, Wednesday](#id-section38)
    * [Entry 39: 2020-03-05, Thursday](#id-section39)
    * [Entry 40: 2020-03-06, Friday](#id-section40)
    * [Entry 41: 2020-03-09, Monday](#id-section41)
    * [Entry 42: 2020-03-10, Tuesday](#id-section42)
     * [Entry 43: 2020-03-11, Wednesday](#id-section43)
    * [Entry 44: 2020-03-12, Thursday](#id-section44)
    * [Entry 45: 2020-03-13, Friday](#id-section45)
    * [Entry 46: 2020-03-16, Monday](#id-section46)
    * [Entry 47: 2020-03-17, Tuesday](#id-section47)
     * [Entry 48: 2020-03-18, Wednesday](#id-section48)
    * [Entry 49: 2020-03-19, Thursday](#id-section49)
    * [Entry 50: 2020-03-20, Friday](#id-section50)
    * [Entry 51: 2020-03-23, Monday](#id-section51)
     * [Entry 52: 2020-03-24, Tuesday](#id-section52)
     * [Entry 53: 2020-03-25, Wednesday](#id-section53)
     * [Entry 54: 2020-03-26, Thursday](#id-section54)
    * [Entry 55: 2020-03-27, Friday](#id-section55)
    * [Entry 56: 2020-03-30, Monday](#id-section56)
     * [Entry 57: 2020-03-31, Tuesday](#id-section57)
     * [Entry 58: 2020-04-01, Wednesday](#id-section58)
     * [Entry 59: 2020-04-02, Thursday](#id-section59)
    * [Entry 60: 2020-04-03, Friday](#id-section60)
    * [Entry 61: 2020-04-06, Monday](#id-section61)
     * [Entry 62: 2020-04-07, Tuesday](#id-section62)
     * [Entry 63: 2020-04-08, Wednesday](#id-section63)
    * [Entry 64: 2020-04-09, Thursday](#id-section64)
    * [Entry 65: 2020-04-10, Friday](#id-section65)
    * [Entry 66: 2020-04-13, Monday](#id-section66)
    * [Entry 67: 2020-04-14, Tuesday](#id-section67)
    * [Entry 68: 2020-04-15, Wednesday](#id-section68)
     * [Entry 69: 2020-04-16, Thursday](#id-section69)
    * [Entry 70: 2020-04-17, Friday](#id-section70)
    * [Entry 71: 2020-04-20, Monday](#id-section71)
    * [Entry 72: 2020-04-21, Tuesday](#id-section72)
    * [Entry 73: 2020-04-22, Wednesday](#id-section73)
    * [Entry 74: 2020-04-23, Thursday](#id-section74)
    * [Entry 75: 2020-04-24, Friday](#id-section75)
    * [Entry 76: 2020-04-27, Monday](#id-section76)
     * [Entry 77: 2020-04-28, Tuesday](#id-section77)
     * [Entry 78: 2020-04-29, Wednesday](#id-section78)
    * [Entry 79: 2020-04-30, Thursday](#id-section79)
    * [Entry 80: 2020-05-01, Friday](#id-section80)
    * [Entry 81: 2020-05-04, Monday](#id-section81)
    * [Entry 82: 2020-05-05, Tuesday](#id-section82)
    * [Entry 83: 2020-05-06, Wednesday](#id-section83)
    * [Entry 84: 2020-05-07, Thursday](#id-section84)
    * [Entry 85: 2020-05-08, Friday](#id-section85)

------
<div id='id-section1'/>   
### Entry 1: 2020-01-01, Wednesday.   



------
<div id='id-section2'/>   
### Entry 2: 2020-01-02, Thursday.   



------
<div id='id-section3'/>   
### Entry 3: 2020-01-03, Friday.   



------
<div id='id-section4'/>   
### Entry 4: 2020-01-06, Monday.   



------
<div id='id-section5'/>   
### Entry 5: 2020-01-07, Tuesday.   



------
<div id='id-section6'/>   
### Entry 6: 2020-01-08, Wednesday.   



------
<div id='id-section7'/>   
### Entry 7: 2020-01-09, Thursday.   



------
<div id='id-section8'/>   
### Entry 8: 2020-01-10, Friday.   



------
<div id='id-section9'/>   
### Entry 9: 2020-01-13, Monday.   



------
<div id='id-section10'/>   
### Entry 10: 2020-01-14, Tuesday.   



------
<div id='id-section11'/>   
### Entry 11: 2020-01-15, Wednesday.   



------
<div id='id-section12'/>   
### Entry 12: 2020-01-16, Thursday.   



------
<div id='id-section13'/>   
### Entry 13: 2020-01-17, Friday.   



------
<div id='id-section14'/>   
### Entry 14: 2020-01-20, Monday.   



------
<div id='id-section15'/>   
### Entry 15: 2020-01-21, Tuesday.   



------
<div id='id-section16'/>   
### Entry 16: 2020-01-22, Wednesday.   
#### Class server: [link](pbio381.uvm.edu)
-Secure shell (ssh), provides secure connection between local machine and server
-copy shared data into personal server (lkshapir@pbio381)
-Focus on edge populations of red spruce (*Picea rubens*), extract rows of data that correspond to edge pops (use "grep" for "e" regions- searches for particular set of characters or numbers, make sure it matches the whole cell: "grep_-w_"E"_filename)



------
<div id='id-section17'/>   
### Entry 17: 2020-01-23, Thursday.   



------
<div id='id-section18'/>   
### Entry 18: 2020-01-24, Friday.   



------
<div id='id-section19'/>   
### Entry 19: 2020-01-27, Monday.   



------
<div id='id-section20'/>   
### Entry 20: 2020-01-28, Tuesday.   

------
<div id='id-section21'/>   
### Entry 21: 2020-01-29, Wednesday.   

-Use .md file instead of .Rmd for this notebook 

-Use three "back ticks" to start a code block ("```")

```
code block 
```

For example:

```
cd ~\mydata
```

Red Spruce (*Picea rubens*):

* High elevations in Vermont, and at higher elevations further south 
* Forms "sky islands" in southern Appalachians 
* Interest in fragmented edge pops, remained south while other species tracked glacial retreat 
* Northern pops are relatively young (emerged after retreat of Laurentide ice sheet)
* Where are alleles being selected for located? Do we need to conserve alleles in fragmented pops declining as a result of changing climate 
* Monitor how current conservation efforts are taking genetics of re-planted trees into account - **Are there particular seed sources that will maximize the genetic diversity of this species? Can we look for pops that are most climate adapted?**



"Pipeline" for data Analysis:

* Visualize quality of raw fast.gz data (**FastQC**)- is it 150 bp in each direction, how is the quality score?

* Go back to original fast.gz file and trim to clean (**Trimmomatic**), visualize again 

* Map/ Align reads on trim file to reference genome (**bwa**) 

  * Input= .fastq, Output= .sam file

* Alignment post-processing: **samtools**, **sambamba**- converts alignment into binary form (.bam file)

* Remove duplicate PCR reads, calculate alignment stats 

  

Sequence code for fastq files: Population_Mother Tree_Sequence file 1 or 2 (from forward or reverse exome reads)

open .gz file:

```zcat 
zcat filename
```

* Starts with @ symbol, gives info on sequencer, where the read was taken from 

* Barcode is first line (w/ plus in the middle)
* Second line is actual sequence (with N's), expect 150 for 150 bp reads
* Third line is a separator between sequence and score 
* Fourth line gives us quality score (like a chromatogram)- 

Quality score interpretation:

```
 Quality encoding: !"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHI
                   |         |         |         |         |
    Quality score: 0........10........20........30........40   
```

"Phred quality score"- high quality scores  have a higher chance that the machine correctly called the nucleotide (ex: quality score of 2 is a 20% chance of being wrong)

Assigned Pop: **XDS**

***Vim editing:***

```.vim```

* Start with ```#!``` ("shablang")

* Click "i" to start editing
* Exit using escape
* ```:wq + filename``` to save

```chmod```= used to change permissions on a file 

-Run FastqC on my XDS sequences from paired clean reads 

```
cd /data/project_data/RS_ExomeSeq/fastq/edge_fastq/pairedcleanreads/
```



------
<div id='id-section22'/>   
### Entry 22: 2020-01-30, Thursday.   



------
<div id='id-section23'/>   
### Entry 23: 2020-01-31, Friday.   



------
<div id='id-section24'/>   
### Entry 24: 2020-02-03, Monday.   



------
<div id='id-section25'/>   
### Entry 25: 2020-02-04, Tuesday.   



------
<div id='id-section26'/>   
### Entry 26: 2020-02-05, Wednesday.   

*Picea albens*

* Very large genome (19.6GB 1n)
* 24 chromosomes (12 1n)
* N50 = smallest contig at which the size of contigs that size and larger give you half the size of the total genome (half of the genome is assembled into large chunks): all of the contigs 101,375 bp and larger make up half of the genome 

**Kill command in terminal:** `Ctrl-C`

Pipeline, mapping and process files located in "myscripts" in repo - used 

Run pipeline w/ mapping and process. sh files incorporated

`bash mypipeline.sh `

* Make sure to copy in pipeline commands before running!

screen: runs commands in background

`Ctrl-A + D` to detach from screen 

check bwa progress on screen:

`screen -r`





------
<div id='id-section27'/>   
### Entry 27: 2020-02-06, Thursday.   



------
<div id='id-section28'/>   
### Entry 28: 2020-02-07, Friday.   



------
<div id='id-section29'/>   
### Entry 29: 2020-02-10, Monday.   

***Genome-Wide Association Studies***(GWA's) - Info Update 

* First study 2002 on myocardial infraction (beginnings in human genetics)

Examples of GWA's in medicine

* Complex genetic diseases (low inheritance)
  * Diabetes, Parkinson's disease, heart abnormalities, Crohn's disease

*Definition of GWA*

* Whole-genome SNP analysis (case control)- case population vs. control -comparison of genetic variants
* Which SNP's are most likely to be important given differences between two groups 
* See if variants are connected to a particular trait 
* Clusters of data points on Manhattan plots shows high association with particular position of SNP's and the trait you are analyzing = possible locations of responsible genes 
* Higher difference between populations, higher peak on plot will be (more significant difference) 

*GWA methodology*

1. Making/finding populations that differ by a singular trait
2. Using DNA techniques to capture data on genetic variants
3. Making statistical inferences on nature of change 

**Three potential sources of genetic differences:**

		1. Causal impact of genetic variants (*great*)
  		2. Close to causal genetic variants (due to crossover etc..) (*good*)
                		3. Bias/Population bias (incorrect or messy sampling) (*bad*)

*Problems with GWA's*

* SNP's are not genes! (differences might not mean anything)
* Complicated when heritability is low - distinctions between samples might become unclear, increased margin of error 
* Ancestral association might cloud results- issues in smaller populations (contradictions to Wright-Fisher pops)

How do we address these issues?

*Fitting the data*

$$\lambda_{expected}= (Median(Empirically observed observation)/ Median(expected)$$

1. Correlated traits > joint GWAs analyses, principal component analysis (PCA) Genomic Control 

2. Trait-structured extrapolation > SNP's > (Effective size, allele freq's)> phenotype
   1. Ability to detect difference depends on population size and effect size of that allele 

------
<div id='id-section30'/>   
### Entry 30: 2020-02-11, Tuesday.   



------
<div id='id-section31'/>   
### Entry 31: 2020-02-12, Wednesday.   

See your own .sam files in BWA file:

` ll XDS*sam`

.sam= sequence alignment file 

Each row in the .sam file is a name of the contig found in the reference genome, line next to it is it's length 

One row of data= one of the pair of reads mapped somewhere on the genome

**File Name**

```GWNJ-0842:368:GW1809211440:2:2224:29467:72965   163     MA_50874        6853    10      7S53M3S =       6987    283     CNAGGGCCANCGNGCNGATACTCTCCTTNGGCCAAGGATTGAGTAGCCNAGAACACAAACNNT    A#AFFJJJJ#JJ#JJ#JJJJJJJJFJJJ#JJJFFJJJJJJFJJJFJJJ#JJ<JJJJJJJJ##F    NM:i:7  MD:Z:2G2A2A11C0T1A17T11 AS:i:33 XS:i:36 ```

* GWNJ...= name of file 
* 163 = flag (how the read behaves during mapping): use http://broadinstitute.github.io/picard/explain-flags.html to explain meaning of flag numbers
* 10 = base quality score (higher quality scores are better)

Get stats on .sam files > use `samtools flagstat filename`

Use loop to get stats on all XDS files - calculate  using `filename sorted rmdup.bam` files + `.bai` file (index)

* `awk` - strips out only files that we need

Use ANGSD to calculate nucleotide diversity (Watterson's theta [# of segregating sites], Tajima's D), $F_{st}$ between populations from genotype likelihoods (GL's) 

------
<div id='id-section32'/>   
### Entry 32: 2020-02-13, Thursday.   



------
<div id='id-section33'/>   
### Entry 33: 2020-02-14, Friday.   



------
<div id='id-section34'/>   
### Entry 34: 2020-02-17, Monday.   



------
<div id='id-section35'/>   
### Entry 35: 2020-02-18, Tuesday.   



------
<div id='id-section36'/>   
### Entry 36: 2020-02-19, Wednesday.   

Estimating rough SFS > `.sfs` > Estimate SFS using `.sfs` as a prior (more robust)> doTheta ($\theta$) > theta_stats 

Tajima's D shows that the bottleneck occured before the populations were fragmented (they all share a similar positive value)

- check reads from flagstat (2.5 million reads with 60% mapping)

------
<div id='id-section37'/>   
### Entry 37: 2020-02-20, Thursday.   



------
<div id='id-section38'/>   
### Entry 38: 2020-02-21, Friday.   



------
<div id='id-section39'/>   
### Entry 39: 2020-02-24, Monday.   

**Transcriptomics**

* Ecological Genomics: Population genetics (p+q, Coalescent theory) + Molecular Evolution (Tajima's D, $\theta$, SFS, $\pi$), Ecology + Next Gen Sequencing Methods

Transcriptomics at epicenter of interacting factors 

​	Experimentally hold/control envrionmental conditions constant to reveal genetically controlled differences of phenotypes at the molecular level (expression)

*Phenotypes*

​	-Characterisitics of an organism that are observable 

Examples, observable at different scales:

* Color (morphological scale)
* Disease resistance/ suseptibility (@ cellular/molecular level)

*Central Dogma of Molecular Biology*:

DNA -(transcription)> RNA -(translation)> Protein --> **Phenotype** (natural selection acts on)

*Reverse ecology*: Genetic signatures describe phenotype natural selection is acting on 

-Changes in phenotype:

Mutations! 

* Mutations in DNA coding regions that alter proteins
* Splice sit variation 
* Promoters and enhancers 
  * Expression level 
  * Timing of expression 
  * cell/tissue type where expressed 
  * Conditions under which a gene is expressed

Post-translational modifications (phosphorylation)

Epigenetic modifications 

* Methylation, histone modification etc..

Transcriptomics focuses on population of RNA in a cell after transcription 

* gives active picture of what is expressed in a cell at a given time under particular conditions 
* global unbiased picture of material (qPCR)
* much cheaper to focus on RNA than [for example] proteomics studies (Illumina)
* de novo transcriptome 

P = G + E + (GxE)

-In transcriptomics, E is controlled so information on G can be clearly extracted

-<u>Experimentally induce</u> your phenotype of interest

​				**-> ** Molecular underpinnings of our phenotype 



Question-hypothesis 

**<u>Factors</u>**

* Common garden 
  * Controlling environmental conditions
* Treatments/ conditions you are interested in studying 
  * related to phenotype of interest
* Populations 
* Tissue, cell type (expression may vary depending)
* Life History stage
* Transgenerational environment 
* Sex/ reproductive stage

*Workflow*

1. Careful experimental design 

   1. Question, hypotheses, broad 'omics, *a priori* hyp.
   2. Factors

2. Experiment

   1. Sample tissues and individuals 
   2. Save RNA quickly! 
   3. Extract, prepare, sequence (whole mRNA, miRNA, 3' tag [good for magnitude of expression])

3. Process and analyze data

   1. Check quality, clean, recheck quality (working with `.fastq` files)
2. Map to reference, extract count data
   3. Normalize for variation among samples (for some that have less or more data) > forms count data matrix, associated with factors (population, tissue, etc...) - test association between expression and factors using ANOVA 
4. Integrate: GO, Network analyses, SNP, TajD, microbiome, epigenetic...



------
<div id='id-section40'/>   
### Entry 40: 2020-02-25, Tuesday.   



------
<div id='id-section41'/>   
### Entry 41: 2020-02-26, Wednesday.   



------
<div id='id-section42'/>   
### Entry 42: 2020-02-27, Thursday.   



------
<div id='id-section43'/>   
### Entry 43: 2020-02-28, Friday.   



------
<div id='id-section44'/>   
### Entry 44: 2020-03-02, Monday.   

<u>**Sea Cucumber Discussion Notes**</u>

Intro:

This paper chose to look at the expression response to change in hydrostatic pressure marine invertebrate (*Apostichopus japonicus)*. Because of the effects of anthropogenic climate change, many shallow water marine invertebrates (live in gravel beds slighly offshore) may be forced to greater oceanic depths as a response to increasing surface (and deeper temperature) shifts. Typically we think of organisms contracting or changing their ranges in response to climate in terms of latitude (or altitude, which is a similar terrestrial parallel) but not bathymetry, which has an added element that these organisms must respond to large changes in pressure, along with oxygen content. In addition, we also don't really consider the physical (in terms of actual physics)(or mechanical) effects of pressure on cells and therefore how this might in turn affect expression- the authors discuss the strengthening of hydrogen bonds under pressure- which may affect processes related to genetic structure (replication, tranlation, etc..) or how the membrane might responsd to pressure. In fact, organisms that are deep-sea adapted are though to have experienced a population bottleneck imposed by physiological limitations in temeperature and pressure, and these factors in conjunction likely limit deeper water transitions of shallow groups.  This paper attempts to look at DEG's (differentially expressed genes) and pinpoint those expressed and attempt to connect them to their specific functional pathways/make functional claims to unveil molecular mechanisms by which these cucumbers might be able to acclimatize (or not) depending possibly on whether they exhibit some sort of phyisiological plasticity in terms of depth response, and they wanted to use a model organism that had both known shallow and deep water members. They compare the expression of DEG's between different pressure treatments (so 25 MPa (246 Atm- depth at and 2500 m: midnight zone),15 Mpa (148 Atm-depth at 1500 m: twilight zone) to a control group not under pressure (0.1MPa=1atm)), and categorize them based on expression patterns (categorized into LRG's, PSG's, and HPG's. We are focusing on the expression response to this deep sea immersion, which genes are up-regulated or down-regulated in this process..etc.... **At the cellular level increases in hydrostatic pressure can have drastic structural and metabolic consequences, including changes in membrane composition- the regulation of these elements may therefore be something that can be seen the differential expression of certain genes**. They used a widespread temperature shallow sea cucumber in this study- *Apostichopus japonicus* *, because it was assumed it contained a homologous gene set to other deep sea sea cucumbers and had the potential to acclimatize (or else they would have just killed off all of their subjects). An important distinction I think is that they are not saying this species might as well be a deep sea species, but rather that it has a chance to *acclimatize*. 

**Methods**

Their methods section is a little messy, they seem to have split it up kind of funny.... So they collected their specimens from an aquaculture farm. First I think this is a little sketchy- 

* I think that farmed specimens might show some inprint of artificial selection in their transcriptome if these were being farmed for food- differential gene expression between farmed/ natural pops? Has been shown in salmon . 
* Provided standard salinity and light, fed with algae. Experimental individuals were starved as to tease out metabolism pathwaysconcerning food intake (?)- 
* Tried to test @ 35Mpa, but individuals died so they used 15 MPa or 25 
*  Acclimatized to recover from stress for two weeks 
* Temperature was maintained at 15 degrees (within ideal range for this species) in a water bath in the pressure container
* Placed three cucumbers in water chamber, pressure was increased at 1 MPa per minute to either experimental pressure (15 or 25 MPa (148 or 246 Atm))- what depths do these correspond to?
* Pressure released immediately after 24 hours, they were flash frozen immediately in liquid nitrogen- stored for RNA extraction
* Used body wall tissue for RNA extraction (I don't know much on sea cucumber anatomy/ tagmata like last week), utilized RNA-Seq for library prep and extraction, sequenced using Hi-Seq, generated 150 bp paired end reads
* Transcripts were assembled using Trinity, checked using BUSCO, unigenes were identified and annotated into databases
* Unigenes annotated using KAAS (KEGG Automatic Annotation Server) + diamond +HMMER > annotates functional genes via BLAST amino acid comparison
* Calculated FPKM (unit of gene expression that normalizes depth and coverage- "fragments per kilobase of exon per million reads mapped")- for each gene used for paired end reads as opposed to RPKM (reads...) because two reads could be a fragment in paired end....
* **Used DESeq2 to identify DEG's** 
  	1. Estimate gene-wide dispersion (higher for small mean counts) - **DeSEq2 used to estimate DEG's between P15 and P0.1,P25 and P0.1, and P15 and P25**
   	2. Generate estimates of within group variation for each gene (3 individuals in each treatment)
   	3. Share info on variation across genes to get more info (similar expression=similar dispersion)
   	4. Given count values of replicates, most likely estimate of dispersion is calc. 
   	5. Plot expected dispersion values for genes of given expression strength (based on read counts)
   	6. Shrink toward predicted values (increase accuracy in small sample sizes) more samples=less shrinkage
   	7. Low dispersion w/ low read count= brought closer to curve
   	8. Negative binomial distribution fitted for each gene, p values and fold change ID'd > DEG's had a p value of <0.01 (significant) and a  cutoff |fold change| (between treatments) >1 (upregulated or down regulated)
 * Pfam + KEGG Enrichment (=overrepresented in the set) Analysis to identify biological processes/ pathways in each test group
    * Based on functional annotations (using KAAS) of genes 
    * KOBAS system Generates a set of statistically enriched or common pathways related to input gene pulled from KEGG data (databases of known, drawn pathways) 
    * Used Fisher's exact test (test for nonrandom assocation between two variables) for testing statistical significance (or else there would be a high chance any of these DEG's would correspond to a pathway)- only in Pfam
    * Also annotated genes in Pfam (are gene family and protein family analogous?)- separately analyzed each group of DEG's using fisher.test (to find statistically significant (enriched) protein/gene families via Pfam database)

**Results**

-Considered up-regulated genes as activated - only essential (up-regulated) processes can be maintained

***Fig. 1<img src="/Users/lilyshapiro/Desktop/Screen Shot 2020-03-01 at 8.31.38 PM.png" alt="Screen Shot 2020-03-01 at 8.31.38 PM" style="zoom:25%;" />*** 

There were 542 up-regulated genes between P25 and P15, 1375 between P25 and P0.1, and 598 between P15 and P0.1. Linear related DEG's were expressed in all groups (genes that were up-regulated under any pressure increase), Pressure-sensitive DEG's were genes up-regulated in each of the pressure treatments compared to the control, and High Pressure DEG's were genes up-regulated across pressure treatments after transition to high pressure. Other numbers are genes up-regulated in each group independent of pressure

 <img src="/Users/lilyshapiro/Desktop/Screen Shot 2020-03-01 at 8.45.24 PM.png" alt="Screen Shot 2020-03-01 at 8.45.24 PM" style="zoom:25%;" />

**Fig 2**: LRG's were linearly up-regulated with increasing pressure (between pressure treatments or with comparisons with the control). Points here represent the fold change in expression between all genes (so all were positively up-regulated)- Change in expression from 15 to 25 is balanced off, although these genes were still up-regulated (not as much). Smaller change in expression from 15 to 25 mPa in pressure-sensitive DEG's. High pressure- induced genes showed little expression up to 15 MPa, but then were highly upregulated to 25 MPa (expression triggered by high pressure). *Where is the shift boundary?*

![Screen Shot 2020-03-01 at 9.38.40 PM](/Users/lilyshapiro/Desktop/Screen Shot 2020-03-01 at 9.38.40 PM.png)

**Fig 3:** Heatmaps of DEG's that were annotated in Swiss-Prot. Red = highly upregulated. Genes associated with LRG's (linearly up-regulated w/ pressure, can see based on temperature change in heatmap) +Genes associated (mostly homeostasis), DEG's associated with PSG's (pressure-sensitive), little change between P15 + P25, mostly homeostasis + signal transduction, + DEG's associated with High pressure, all significantly up-regulated at P25 (wow!), mostly annotated for homeostasis and genetic info processing

**LRG's** (14/38) annotated in Swiss-Prot database associated with

1.  **homeostasis maintenance** [3=ubiquitination[marks for degradation, movement],heat shock protein,antioxidative, kinase for innate immunity], 
2. **lysosomal activities** 
3. energy metabolism
4.  lipid metabolism

Is LRG biased b/c it includes strong difference in expression for HPG's??

**PSG's** (70/244) annotated in Swiss-Prot database associated with

1. **Homeostasis maintenance** (15) [6=Stress response (DNA Damage response, unfolded protein response, 6=Immnune response, 2=apoptosis, 3=ubiquitination]
2. **Signal transduction** (15) [7 in transcription]
3. **Genetic information processing** (12)
4. Lysosomal activities (5)
5. Lipid metabolism (2)
6. Other (16)

**HPG's** (123/257 annotated) in Swiss Prot database associated with > **most annotated, what is this a factor of?**

1. **Homeostasis maintenance** (23) [13=ubiquitination, 8=stress response]
2. **Genetic Information Processing** (22) [12=transcription, 7=translation]
3. Signal transduction
4. Lysosomal activities
5. Membrane-related function
6. Lipid metabolism
7. Others (unidentified?)

**More effort to maintain homeostasis under stress= more upregulation of homeostatic genes**

**Similar up-regulation of genes associated with particular processes across DEG's (homeostasis, antioxidation,stress response)**

**Genes involved in ubiquitination pathways are very diverse, hard to pin down to deep-sea expression/acclimatization responses**

Interesting to me that there wasn't a greater response to stress. Also, this experiment could be tested at different life stages? Or with other organisms that also have deep-sea counterparts? 

![Screen Shot 2020-03-01 at 10.15.11 PM](/Users/lilyshapiro/Desktop/Screen Shot 2020-03-01 at 10.15.11 PM.png)

Fig 4: we don't have to go over the whole pathway, but basically this is the only statistically enriched pathway the author's found for all of the DEG groups combined-so 17 of the total DEG's associated with pressure were found to correspond with this molecular pathway in the KEGG database, which is and endocytotic pathway. The Red boxes represent the genes upregulated  (corresponding proteins in supplemental materials) at high pressure 

Diverse functionality in cells, endocytosis found to be significant pathway in HPG's, and in all subdivisions of DEG's  > **Endocytosis facilitates faster development of a symbiotic system in deep-adapted organisms (? don't know if this is too specific)** Ubiquitination + Endocytosis may act together to regulate elimination of misfolded proteins

![Screen Shot 2020-03-01 at 11.26.38 PM](/Users/lilyshapiro/Desktop/Screen Shot 2020-03-01 at 11.26.38 PM.png)

**Fig. 5:**Displaying results of gene/ protein family analysis (Pfam enrichment). Gene families that are significantly associated with each of these DEG categories. 13 gene families enriched in LRG's, 13 gene families enriched in PSG's (about half of these are involved in transcription), and 20 gene families (!) enriched in HPG's (6 of which are involved in genetic info functionality- graph's show the number of genes in each family (circles, colored by p-value). The list of families is on the y axis- genes are grouped in families based on the function of the proteins they encode for. Rich factor is bascially describing the representation of the DEG's in that particular family out of all of the genes in that family- so higher numbers mean that particular group of DEG's is more enriched, more represented. YOu can see that there are more groups within the HPG's that are enriched (and more gene families overall)- maybe these gene families are smaller overall? Or not because the x axis is scaled smaller? 

**Gene families having to do with genetic information, particularly transcirption are overall enriched- high pressure strengthens H-bonds and can impede genetic info processes, these might be upregulated- found to be positively selected in other deep-sea organisms, might also need high frequencies of DNA repair**

I think the physical effects of deep sea compression on physiology and the genomic response to this is super interesting and not usually considered (the actual physics) affecting DNA bonds or membrane structure (actual cellular physical transformation) that might lead to a change in expression. I don't think there have been many studies of this kind, this one is brand new (2020), and I think it's an interesting future prospect. I would like to see more studies on various groups, and maybe comparing different genera of the same family to see differential responses. Also comparison of temperature to tropical species, at different life stages, etc....long term studies. How can we streamline transcriptomics/ gene orthology studies? Combined effect of temp and pressure looking for genes involved in preventing anoxia? 

<div id='id-section45'/>   
### Entry 45: 2020-03-03, Tuesday.   



------
<div id='id-section46'/>   
### Entry 46: 2020-03-04, Wednesday.   

**Coding work**- Pops by treatment: ESC_01_C*

-Create a for-loop so `salmon` can quantify transcript abundance + compare them to a reference genome (*P. abies*)

- Set working directory to location of cleaned reads
- For each file in the cleaned reads directory, run salmon 
- Set reference genome after -i
- Set reads after -r, specify directory to clean reads, follow with specific file
- Output to salmon/cleanedreads
- Script to run salmon found in:  `~/Documents/Github/Ecological-Genomics/myscripts/salmon.sh`

Output of salmon: Mapping rate to reference (in `.log` file)

Mapping rate for ESC_01_C_0_CAGATG_R1.cl.fq = **3.16742%**

<u>*new Mapping Rate (w/ all reference): **33.6983%***</u>

Mapping rate for ESC_01_C_10_ACAACG_R1.cl.fq= **2.89247%** *(% of reads mapped to reference with high confidence, based on homology from BLAST between red-spruce and white spruce)*

<u>*new Mapping Rate: **28.6074%***</u>

* Mapped to high quality reference genome, lowered `kmer` parameter to allow mapping of smaller fragments, need to use unadjusted reference (no lowering of `kmer` to allow small fragment mapping)
* Changed output directory to "allmapping", changed reference to "Pabies_cds_index"

> Completed in bash using R code (not RStudio)

```
library(tximportData)
library(tximport)

#locate the directory containing the files. 
dir <- "/data/project_data/RS_RNASeq/salmon/"
list.files(dir) #Lists files in directory

# read in table with sample ids
samples <- read.table("/data/project_data/RS_RNASeq/salmon/RS_samples.txt", header=TRUE) #Use to line up sample names with quant.sf files

# now point to quant files
all_files <- file.path(dir, samples$sample, "quant.sf")
names(all_files) <- samples$sample

# what would be used if linked transcripts to genes
#txi <- tximport(files, type = "salmon", tx2gene = tx2gene)
# to be able to run without tx2gene
txi <- tximport(all_files, type = "salmon", txOut=TRUE)  
names(txi) #We used this, had not linked transcripts to genes 

head(txi$counts)

countsMatrix <- txi$counts
dim(countsMatrix)
#[1] 66069    76

# To write out
write.table(countsMatrix, file = "RS_countsMatrix.txt", col.names = T, row.names = T, quote = F) 
```

------
<div id='id-section47'/>   
### Entry 47: 2020-03-05, Thursday.   



------
<div id='id-section48'/>   
### Entry 48: 2020-03-06, Friday.   



------
<div id='id-section49'/>   
### Entry 49: 2020-03-09, Monday.   



------
<div id='id-section50'/>   
### Entry 50: 2020-03-10, Tuesday.   



------
<div id='id-section51'/>   
### Entry 51: 2020-03-11, Wednesday.   



------
<div id='id-section52'/>   
### Entry 52: 2020-03-12, Thursday.   



------
<div id='id-section53'/>   
### Entry 53: 2020-03-13, Friday.   



------
<div id='id-section54'/>   
### Entry 54: 2020-03-16, Monday.   



------
<div id='id-section55'/>   
### Entry 55: 2020-03-17, Tuesday.   



------
<div id='id-section56'/>   
### Entry 56: 2020-03-18, Wednesday.   



------
<div id='id-section57'/>   
### Entry 57: 2020-03-19, Thursday.   



------
<div id='id-section58'/>   
### Entry 58: 2020-03-20, Friday.   



------
<div id='id-section59'/>   
### Entry 59: 2020-03-23, Monday.   



------
<div id='id-section60'/>   
### Entry 60: 2020-03-24, Tuesday.   



------
<div id='id-section61'/>   
### Entry 61: 2020-03-25, Wednesday.   



------
<div id='id-section62'/>   
### Entry 62: 2020-03-26, Thursday.   



------
<div id='id-section63'/>   
### Entry 63: 2020-03-27, Friday.   



------
<div id='id-section64'/>   
### Entry 64: 2020-03-30, Monday.   



------
<div id='id-section65'/>   
### Entry 65: 2020-03-31, Tuesday.   



------
<div id='id-section66'/>   
### Entry 66: 2020-04-01, Wednesday.   



------
<div id='id-section67'/>   
### Entry 67: 2020-04-02, Thursday.   



------
<div id='id-section68'/>   
### Entry 68: 2020-04-03, Friday.   



------
<div id='id-section69'/>   
### Entry 69: 2020-04-06, Monday.   



------
<div id='id-section70'/>   
### Entry 70: 2020-04-07, Tuesday.   



------
<div id='id-section71'/>   
### Entry 71: 2020-04-08, Wednesday.   



------
<div id='id-section72'/>   
### Entry 72: 2020-04-09, Thursday.   



------
<div id='id-section73'/>   
### Entry 73: 2020-04-10, Friday.   



------
<div id='id-section74'/>   
### Entry 74: 2020-04-13, Monday.   



------
<div id='id-section75'/>   
### Entry 75: 2020-04-14, Tuesday.   



------
<div id='id-section76'/>   
### Entry 76: 2020-04-15, Wednesday.   



------
<div id='id-section77'/>   
### Entry 77: 2020-04-16, Thursday.   



------
<div id='id-section78'/>   
### Entry 78: 2020-04-17, Friday.   



------
<div id='id-section79'/>   
### Entry 79: 2020-04-20, Monday.   



------
<div id='id-section80'/>   
### Entry 80: 2020-04-21, Tuesday.   



------
<div id='id-section81'/>   
### Entry 81: 2020-04-22, Wednesday.   



------
<div id='id-section82'/>   
### Entry 82: 2020-04-23, Thursday.   



------
<div id='id-section83'/>   
### Entry 83: 2020-04-24, Friday.   



------
<div id='id-section84'/>   
### Entry 84: 2020-04-27, Monday.   



------
<div id='id-section85'/>   
### Entry 85: 2020-04-28, Tuesday.   



------
<div id='id-section86'/>   
### Entry 86: 2020-04-29, Wednesday.   



------
<div id='id-section87'/>   
### Entry 87: 2020-04-30, Thursday.   



------
<div id='id-section88'/>   
### Entry 88: 2020-05-01, Friday.   



------
<div id='id-section89'/>   
### Entry 89: 2020-05-04, Monday.   



------
<div id='id-section90'/>   
### Entry 90: 2020-05-05, Tuesday.   



------
<div id='id-section91'/>   
### Entry 91: 2020-05-06, Wednesday.   



------
<div id='id-section92'/>   
### Entry 92: 2020-05-07, Thursday.   



------
<div id='id-section93'/>   
### Entry 93: 2020-05-08, Friday.   



------
<div id='id-section94'/>   
### Entry 94: 2020-05-11, Monday.   



------
<div id='id-section95'/>   
### Entry 95: 2020-05-12, Tuesday.   



------
<div id='id-section96'/>   
### Entry 96: 2020-05-13, Wednesday.   



------
<div id='id-section97'/>   
### Entry 97: 2020-05-14, Thursday.   



------
<div id='id-section98'/>   
### Entry 98: 2020-05-15, Friday.   



------
<div id='id-section99'/>   
### Entry 99: 2020-05-18, Monday.   



------
<div id='id-section100'/>   
### Entry 100: 2020-05-19, Tuesday.   



------
<div id='id-section101'/>   
### Entry 101: 2020-05-20, Wednesday.   



------
<div id='id-section102'/>   
### Entry 102: 2020-05-21, Thursday.   



------
<div id='id-section103'/>   
### Entry 103: 2020-05-22, Friday.   



------
<div id='id-section104'/>   
### Entry 104: 2020-05-25, Monday.   



------
<div id='id-section105'/>   
### Entry 105: 2020-05-26, Tuesday.   



------
<div id='id-section106'/>   
### Entry 106: 2020-05-27, Wednesday.   



------
<div id='id-section107'/>   
### Entry 107: 2020-05-28, Thursday.   



------
<div id='id-section108'/>   
### Entry 108: 2020-05-29, Friday.   



------
<div id='id-section109'/>   
### Entry 109: 2020-06-01, Monday.   



------
<div id='id-section110'/>   
### Entry 110: 2020-06-02, Tuesday.   



------
<div id='id-section111'/>   
### Entry 111: 2020-06-03, Wednesday.   



------
<div id='id-section112'/>   
### Entry 112: 2020-06-04, Thursday.   



------
<div id='id-section113'/>   
### Entry 113: 2020-06-05, Friday.   



------
<div id='id-section114'/>   
### Entry 114: 2020-06-08, Monday.   



------
<div id='id-section115'/>   
### Entry 115: 2020-06-09, Tuesday.   



------
<div id='id-section116'/>   
### Entry 116: 2020-06-10, Wednesday.   



------
<div id='id-section117'/>   
### Entry 117: 2020-06-11, Thursday.   



------
<div id='id-section118'/>   
### Entry 118: 2020-06-12, Friday.   



------
<div id='id-section119'/>   
### Entry 119: 2020-06-15, Monday.   



------
<div id='id-section120'/>   
### Entry 120: 2020-06-16, Tuesday.   



------
<div id='id-section121'/>   
### Entry 121: 2020-06-17, Wednesday.   



------
<div id='id-section122'/>   
### Entry 122: 2020-06-18, Thursday.   



------
<div id='id-section123'/>   
### Entry 123: 2020-06-19, Friday.   



------
<div id='id-section124'/>   
### Entry 124: 2020-06-22, Monday.   



------
<div id='id-section125'/>   
### Entry 125: 2020-06-23, Tuesday.   



------
<div id='id-section126'/>   
### Entry 126: 2020-06-24, Wednesday.   



------
<div id='id-section127'/>   
### Entry 127: 2020-06-25, Thursday.   



------
<div id='id-section128'/>   
### Entry 128: 2020-06-26, Friday.   



------
<div id='id-section129'/>   
### Entry 129: 2020-06-29, Monday.   



------
<div id='id-section130'/>   
### Entry 130: 2020-06-30, Tuesday.   



------
<div id='id-section131'/>   
### Entry 131: 2020-07-01, Wednesday.   



------
<div id='id-section132'/>   
### Entry 132: 2020-07-02, Thursday.   



------
<div id='id-section133'/>   
### Entry 133: 2020-07-03, Friday.   



------
<div id='id-section134'/>   
### Entry 134: 2020-07-06, Monday.   



------
<div id='id-section135'/>   
### Entry 135: 2020-07-07, Tuesday.   



------
<div id='id-section136'/>   
### Entry 136: 2020-07-08, Wednesday.   



------
<div id='id-section137'/>   
### Entry 137: 2020-07-09, Thursday.   



------
<div id='id-section138'/>   
### Entry 138: 2020-07-10, Friday.   



------
<div id='id-section139'/>   
### Entry 139: 2020-07-13, Monday.   



------
<div id='id-section140'/>   
### Entry 140: 2020-07-14, Tuesday.   



------
<div id='id-section141'/>   
### Entry 141: 2020-07-15, Wednesday.   



------
<div id='id-section142'/>   
### Entry 142: 2020-07-16, Thursday.   



------
<div id='id-section143'/>   
### Entry 143: 2020-07-17, Friday.   



------
<div id='id-section144'/>   
### Entry 144: 2020-07-20, Monday.   



------
<div id='id-section145'/>   
### Entry 145: 2020-07-21, Tuesday.   



------
<div id='id-section146'/>   
### Entry 146: 2020-07-22, Wednesday.   



------
<div id='id-section147'/>   
### Entry 147: 2020-07-23, Thursday.   



------
<div id='id-section148'/>   
### Entry 148: 2020-07-24, Friday.   



------
<div id='id-section149'/>   
### Entry 149: 2020-07-27, Monday.   



------
<div id='id-section150'/>   
### Entry 150: 2020-07-28, Tuesday.   



------
<div id='id-section151'/>   
### Entry 151: 2020-07-29, Wednesday.   



------
<div id='id-section152'/>   
### Entry 152: 2020-07-30, Thursday.   



------
<div id='id-section153'/>   
### Entry 153: 2020-07-31, Friday.   



------
<div id='id-section154'/>   
### Entry 154: 2020-08-03, Monday.   



------
<div id='id-section155'/>   
### Entry 155: 2020-08-04, Tuesday.   



------
<div id='id-section156'/>   
### Entry 156: 2020-08-05, Wednesday.   



------
<div id='id-section157'/>   
### Entry 157: 2020-08-06, Thursday.   



------
<div id='id-section158'/>   
### Entry 158: 2020-08-07, Friday.   



------
<div id='id-section159'/>   
### Entry 159: 2020-08-10, Monday.   



------
<div id='id-section160'/>   
### Entry 160: 2020-08-11, Tuesday.   



------
<div id='id-section161'/>   
### Entry 161: 2020-08-12, Wednesday.   



------
<div id='id-section162'/>   
### Entry 162: 2020-08-13, Thursday.   



------
<div id='id-section163'/>   
### Entry 163: 2020-08-14, Friday.   



------
<div id='id-section164'/>   
### Entry 164: 2020-08-17, Monday.   



------
<div id='id-section165'/>   
### Entry 165: 2020-08-18, Tuesday.   



------
<div id='id-section166'/>   
### Entry 166: 2020-08-19, Wednesday.   



------
<div id='id-section167'/>   
### Entry 167: 2020-08-20, Thursday.   



------
<div id='id-section168'/>   
### Entry 168: 2020-08-21, Friday.   



------
<div id='id-section169'/>   
### Entry 169: 2020-08-24, Monday.   



------
<div id='id-section170'/>   
### Entry 170: 2020-08-25, Tuesday.   



------
<div id='id-section171'/>   
### Entry 171: 2020-08-26, Wednesday.   



------
<div id='id-section172'/>   
### Entry 172: 2020-08-27, Thursday.   



------
<div id='id-section173'/>   
### Entry 173: 2020-08-28, Friday.   



------
<div id='id-section174'/>   
### Entry 174: 2020-08-31, Monday.   



------
<div id='id-section175'/>   
### Entry 175: 2020-09-01, Tuesday.   



------
<div id='id-section176'/>   
### Entry 176: 2020-09-02, Wednesday.   



------
<div id='id-section177'/>   
### Entry 177: 2020-09-03, Thursday.   



------
<div id='id-section178'/>   
### Entry 178: 2020-09-04, Friday.   



------
<div id='id-section179'/>   
### Entry 179: 2020-09-07, Monday.   



------
<div id='id-section180'/>   
### Entry 180: 2020-09-08, Tuesday.   



------
<div id='id-section181'/>   
### Entry 181: 2020-09-09, Wednesday.   



------
<div id='id-section182'/>   
### Entry 182: 2020-09-10, Thursday.   



------
<div id='id-section183'/>   
### Entry 183: 2020-09-11, Friday.   



------
<div id='id-section184'/>   
### Entry 184: 2020-09-14, Monday.   



------
<div id='id-section185'/>   
### Entry 185: 2020-09-15, Tuesday.   



------
<div id='id-section186'/>   
### Entry 186: 2020-09-16, Wednesday.   



------
<div id='id-section187'/>   
### Entry 187: 2020-09-17, Thursday.   



------
<div id='id-section188'/>   
### Entry 188: 2020-09-18, Friday.   



------
<div id='id-section189'/>   
### Entry 189: 2020-09-21, Monday.   



------
<div id='id-section190'/>   
### Entry 190: 2020-09-22, Tuesday.   



------
<div id='id-section191'/>   
### Entry 191: 2020-09-23, Wednesday.   



------
<div id='id-section192'/>   
### Entry 192: 2020-09-24, Thursday.   



------
<div id='id-section193'/>   
### Entry 193: 2020-09-25, Friday.   



------
<div id='id-section194'/>   
### Entry 194: 2020-09-28, Monday.   



------
<div id='id-section195'/>   
### Entry 195: 2020-09-29, Tuesday.   



------
<div id='id-section196'/>   
### Entry 196: 2020-09-30, Wednesday.   



------
<div id='id-section197'/>   
### Entry 197: 2020-10-01, Thursday.   



------
<div id='id-section198'/>   
### Entry 198: 2020-10-02, Friday.   



------
<div id='id-section199'/>   
### Entry 199: 2020-10-05, Monday.   



------
<div id='id-section200'/>   
### Entry 200: 2020-10-06, Tuesday.   



------
<div id='id-section201'/>   
### Entry 201: 2020-10-07, Wednesday.   



------
<div id='id-section202'/>   
### Entry 202: 2020-10-08, Thursday.   



------
<div id='id-section203'/>   
### Entry 203: 2020-10-09, Friday.   



------
<div id='id-section204'/>   
### Entry 204: 2020-10-12, Monday.   



------
<div id='id-section205'/>   
### Entry 205: 2020-10-13, Tuesday.   



------
<div id='id-section206'/>   
### Entry 206: 2020-10-14, Wednesday.   



------
<div id='id-section207'/>   
### Entry 207: 2020-10-15, Thursday.   



------
<div id='id-section208'/>   
### Entry 208: 2020-10-16, Friday.   



------
<div id='id-section209'/>   
### Entry 209: 2020-10-19, Monday.   



------
<div id='id-section210'/>   
### Entry 210: 2020-10-20, Tuesday.   



------
<div id='id-section211'/>   
### Entry 211: 2020-10-21, Wednesday.   



------
<div id='id-section212'/>   
### Entry 212: 2020-10-22, Thursday.   



------
<div id='id-section213'/>   
### Entry 213: 2020-10-23, Friday.   



------
<div id='id-section214'/>   
### Entry 214: 2020-10-26, Monday.   



------
<div id='id-section215'/>   
### Entry 215: 2020-10-27, Tuesday.   



------
<div id='id-section216'/>   
### Entry 216: 2020-10-28, Wednesday.   



------
<div id='id-section217'/>   
### Entry 217: 2020-10-29, Thursday.   



------
<div id='id-section218'/>   
### Entry 218: 2020-10-30, Friday.   



------
<div id='id-section219'/>   
### Entry 219: 2020-11-02, Monday.   



------
<div id='id-section220'/>   
### Entry 220: 2020-11-03, Tuesday.   



------
<div id='id-section221'/>   
### Entry 221: 2020-11-04, Wednesday.   



------
<div id='id-section222'/>   
### Entry 222: 2020-11-05, Thursday.   



------
<div id='id-section223'/>   
### Entry 223: 2020-11-06, Friday.   



------
<div id='id-section224'/>   
### Entry 224: 2020-11-09, Monday.   



------
<div id='id-section225'/>   
### Entry 225: 2020-11-10, Tuesday.   



------
<div id='id-section226'/>   
### Entry 226: 2020-11-11, Wednesday.   



------
<div id='id-section227'/>   
### Entry 227: 2020-11-12, Thursday.   



------
<div id='id-section228'/>   
### Entry 228: 2020-11-13, Friday.   



------
<div id='id-section229'/>   
### Entry 229: 2020-11-16, Monday.   



------
<div id='id-section230'/>   
### Entry 230: 2020-11-17, Tuesday.   



------
<div id='id-section231'/>   
### Entry 231: 2020-11-18, Wednesday.   



------
<div id='id-section232'/>   
### Entry 232: 2020-11-19, Thursday.   



------
<div id='id-section233'/>   
### Entry 233: 2020-11-20, Friday.   



------
<div id='id-section234'/>   
### Entry 234: 2020-11-23, Monday.   



------
<div id='id-section235'/>   
### Entry 235: 2020-11-24, Tuesday.   



------
<div id='id-section236'/>   
### Entry 236: 2020-11-25, Wednesday.   



------
<div id='id-section237'/>   
### Entry 237: 2020-11-26, Thursday.   



------
<div id='id-section238'/>   
### Entry 238: 2020-11-27, Friday.   



------
<div id='id-section239'/>   
### Entry 239: 2020-11-30, Monday.   



------
<div id='id-section240'/>   
### Entry 240: 2020-12-01, Tuesday.   



------
<div id='id-section241'/>   
### Entry 241: 2020-12-02, Wednesday.   



------
<div id='id-section242'/>   
### Entry 242: 2020-12-03, Thursday.   



------
<div id='id-section243'/>   
### Entry 243: 2020-12-04, Friday.   



------
<div id='id-section244'/>   
### Entry 244: 2020-12-07, Monday.   



------
<div id='id-section245'/>   
### Entry 245: 2020-12-08, Tuesday.   



------
<div id='id-section246'/>   
### Entry 246: 2020-12-09, Wednesday.   



------
<div id='id-section247'/>   
### Entry 247: 2020-12-10, Thursday.   



------
<div id='id-section248'/>   
### Entry 248: 2020-12-11, Friday.   



------
<div id='id-section249'/>   
### Entry 249: 2020-12-14, Monday.   



------
<div id='id-section250'/>   
### Entry 250: 2020-12-15, Tuesday.   



------
<div id='id-section251'/>   
### Entry 251: 2020-12-16, Wednesday.   



------
<div id='id-section252'/>   
### Entry 252: 2020-12-17, Thursday.   



------
<div id='id-section253'/>   
### Entry 253: 2020-12-18, Friday.   



------
<div id='id-section254'/>   
### Entry 254: 2020-12-21, Monday.   



------
<div id='id-section255'/>   
### Entry 255: 2020-12-22, Tuesday.   



------
<div id='id-section256'/>   
### Entry 256: 2020-12-23, Wednesday.   



------
<div id='id-section257'/>   
### Entry 257: 2020-12-24, Thursday.   



------
<div id='id-section258'/>   
### Entry 258: 2020-12-25, Friday.   



------
<div id='id-section259'/>   
### Entry 259: 2020-12-28, Monday.   



------
<div id='id-section260'/>   
### Entry 260: 2020-12-29, Tuesday.   



------
<div id='id-section261'/>   
### Entry 261: 2020-12-30, Wednesday.   



------
<div id='id-section262'/>   
### Entry 262: 2020-12-31, Thursday.   